src/user/user.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@nestjs/common'
+import { InjectRepository } from '@nestjs/typeorm'
+import { User } from '../entities/user.entity'
+import { FindConditions, FindOneOptions, Repository } from 'typeorm'
+import { UpdateRelationshipDto, CreateUserDto, GetUserDto } from './dto/user.dto'
+import { DuplicateElementException, InvalidUserTypeException, UserNotFoundException } from './user.exception'
+
+@Injectable()
+export class UserService {
+  constructor(@InjectRepository(User) private userRepository: Repository<User>) {}
+  schemaToDto(schema: User): GetUserDto {
+    //remove sensitive data
```

Comment:
like what I already commented in controller, remove the sensitive data is good. But in my opinion, we should send only the data that frontend want to use.